Tensile/KernelWriterSource.py

```diff
@@ -3311,7 +3311,13 @@ def wait(self, kernel, tPA, tPB, globalRead, localWrite, localRead, comment):
   # SyncThreads
   ##############################################################################
   def syncThreads(self, kernel, comment=""):
-    return self.indent + self.syncStr + " //" + comment + self.endLine
+    if globalParameters["CurrentISA"] == (9,0,8):
+      return self.indent + self.syncStr + " //" + comment + self.endLine
+    else:
+      if kernel["SubGroup0"]*kernel["SubGroup1"]*kernel["LocalSplitU"] != kernel["WavefrontSize"]:
+        return self.indent + self.syncStr + " //" + comment + self.endLine
+      else:
+        return self.indent + self.endLine
```

Comment:
Here, we can also use the same message as KernelWriterAssembly

Reply:
return "// Skip barrier: NumThreads=%s"%(kernel["NumThreads"]) + \
              comment + self.endLine

Reply:
@nakajee updated code. The poor performance on MI100 is caused from old compiler. A newer compiler like rocm 5.2 doesn't have this issue. I removed conditional statement checking against (9,0,8) (MI100.

Reply:
OK. Please also check other suggestions I put.

(1) please use the condition kernel["NumThreads"] > self.kernel["WavefrontSize"]
  use > instead of !=
  use NumThreads
  
  (2)we can also use the same message as KernelWriterAssembly

Reply:
@nakajee Done.

Reply:
@nakajee use the same comments in KernelWriteAssembly.py.